### PR TITLE
Fix tests not setting up JPMS arguments for Gradle process

### PR DESCRIPTION
**Pull request by mlopatkin**
Modern Gradle requires several `--add-opens` switches to be used or weird failures happen. We need to configure the switches manually when attempting to run Gradle inside the client process, but we were missing a few places and did that in an ad-hoc way.

Fixes gradle/gradle-private#4187

### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by mlopatkin** [comment]
@bot-gradle test APT without PTS

**Comment by mlopatkin** [comment]
@bot-gradle test RfR

**Comment by mlopatkin** [comment]
Apparently, I was overly optimistic with assuming that we only use java home from `AvailableJavaHomes`.

**Comment by mlopatkin** [comment]
@bot-gradle test without PTS

File: testing/internal-performance-testing/src/main/groovy/org/gradle/performance/AbstractCrossVersionPerformanceTest.groovy [hunk 1]
```diff
@@ -70,8 +70,6 @@ class AbstractCrossVersionPerformanceTest extends AbstractPerformanceTest {
         runner.addBuildMutator { invocationSettings ->
             new ApplyDevelocityPluginMutator(invocationSettings.projectDir)
         }
-        // Required by GradleLifecycle API used by Develocity
```
Comment (bamboo) [inline testing/internal-performance-testing/src/main/groovy/org/gradle/performance/AbstractCrossVersionPerformanceTest.groovy:73]: 👍

File: testing/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/PerformanceTestJvmOptions.groovy [hunk 2]
```diff
@@ -34,4 +35,13 @@ class PerformanceTestJvmOptions {
 
         return jvmOptions
     }
+
+    static List<String> normalizeGradleJvmOptions(boolean useDaemon, List<String> originalJvmOptions) {
+        if (JavaVersion.current().isJava9Compatible() && !useDaemon) {
```
Comment (bamboo) [inline testing/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/PerformanceTestJvmOptions.groovy:40]: 💅 Flip `&&` operands to be in the same order as the other occurrence so they are more immediately recognizable as the same condition

**Review by bamboo — APPROVED** [review]
LGTM!

**Comment by mlopatkin** [comment]
@bot-gradle test RfN

**Pull request merged**